example/lib/main.dart

```diff
@@ -36,7 +36,7 @@ class MessageListTile extends StatelessWidget {
   Widget build(BuildContext context) {
     String message = 'No message retrieved!';
     if (document != null) {
-      final data = document!.data();
+      final data = document!.data() as Map<String, dynamic>?;
```

Comment:
A cast won't work here since it'll most likely be a `_InternalLinkedHashMap<String, dynamic>`, and using `as` will throw an error. That's why I recommended using `.from()` instead; you can just change the `if` at the line 38 to also check if `document.data() != null` and then properly convert it below.